docs/web-apps/automated-testing/pre-run-executables.md

```diff
@@ -0,0 +1,308 @@
+
+```
+#!/usr/bin/python
+# -*- coding: UTF-8 -*-
+
+from selenium import webdriver
+
+desired_capabilities = webdriver.DesiredCapabilities.CHROME
+desired_capabilities['version'] = 'latest'
+desired_capabilities['platform'] = 'macOS 10.14'
+desired_capabilities['name'] = 'Editing the DNS'
+desired_capabilities['prerun'] = {'executable':'https://raw.githubusercontent.com/albedithdiaz/sauceSupport/master/preRunScripts/EditHostFile.sh', 'background': False }
+
+
+
+remote_url = 'http://SAUCE_USERNAME:SAUCE_ACCESS_KEY@ondemand.saucelabs.com/wd/hub'
+
+
+driver = webdriver.Remote(command_executor = remote_url, desired_capabilities = desired_capabilities)
+driver.implicitly_wait(30)
+
+driver.get('http://www.saucedemo.com')
+title = driver.title
+assert "Swag Labs" in title
+
+driver.quit()
+```
+
+## Creating JSON Objects for Multiple Arguments
+
+If your test script is written in Java, you will need to create a JSON object if you want to include multiple arguments, such as `--silent, -a` with the `prerun` capability.
+
+>**NOTE:** The JSONObject Class/Library
+Make sure your test script imports the JSONObject class/library so you can create the JSON object.
+
+```
+JSONObject obj = new JSONObject();
+obj.put("executable","http://url.to/your/executable.exe");
+LinkedList<String> list = new LinkedList<String>();
+list.add("/S");
+list.add("-a");
+list.add("-q");
+obj.put("args",list);
+obj.put("background","true");
+obj.put("timeout", "240");
+System.out.print(obj);
+```
+
+If your test script is written in C#, you will need to create a Dictionary object with the prerun capability as JSON Objects do not work well with C# projects
+
+>**NOTE:** The C# Dictionary Object
+When creating the Dictionary object, make sure to make it the TKey is of value "string" and the TValue is of value "object"
+
+```
+var obj = new Dictionary<string, object>();
+obj.Add("executable", "http://url.to/your/executable");
+obj.Add("background", true);
+obj.Add("timeout", 120);
+```
+
+When you set the desired capabilities of your test, you will refer to the object you created as the path to the executable, as in this example.
+
+```
+DesiredCapabilities capabilities = new DesiredCapabilities();
+capabilities.setCapability("prerun", obj);
+```
+
+## Running an AutoIt Script
+
+When using Sauce Connect to run local tests on a Windows machine, you may encounter an Integrated Windows Authentication (IWA) dialog, also known as NTLM or Domain authentication. This is because the machine that Sauce Connect is running on is used to look up the host where the site or application under test is located. If the host has IWA enabled, the authentication request will be passed back through Sauce Connect to the Sauce Labs browser. Because there is no registry key available on our Windows virtual machines to disable this authentication, the solution is to create an AutoIT script to respond to the dialog,
```

Comment:
```suggestion
When using Sauce Connect to run local tests on a Windows machine, you may encounter an Integrated Windows Authentication (IWA) dialog, also known as NTLM or Domain authentication. This is because the machine that Sauce Connect is running on is used to look up the host where the site or application under test is located. If the host has IWA enabled, the authentication request will be passed back through Sauce Connect to the Sauce Labs browser. Because there is no registry key available on our Windows virtual machines to disable this authentication, the solution is to create an AutoIT script to respond to the dialog.
```